src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,490 @@
+﻿namespace Spice86.Core.Emulator.InterruptHandlers.Dos.Xms;
+
+using Serilog.Events;
+
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.Extensions;
+using Spice86.Core.Emulator.InterruptHandlers;
+using Spice86.Core.Emulator.InterruptHandlers.Common.MemoryWriter;
+using Spice86.Core.Emulator.InterruptHandlers.Common.RoutineInstall;
+using Spice86.Core.Emulator.Memory;
+using Spice86.Shared.Emulator.Memory;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+/// <summary>
+/// Provides DOS applications with XMS memory.
+/// </summary>
+public sealed class ExtendedMemoryManager : InterruptHandler, IMemoryDevice {
+    private int _a20EnableCount;
+    private readonly LinkedList<XmsBlock> _xmsBlocksLinkedList = new();
+    private readonly SortedList<int, int> _xmsHandles = new();
+    
+    public const ushort CallbackAsmFunctionSegment = 0xD000;
```

Comment:
Are you sure the segment is fixed? I cannot find references to this address on docs.